services/venice-router/src/main/java/com/linkedin/venice/router/api/VeniceDelegateMode.java

```diff
@@ -349,41 +363,130 @@ public <H, P extends ResourcePath<K>, K, R> Scatter<H, P, K> scatter(
-    H host;
-    long minCount = Long.MAX_VALUE;
+  /**
+   * Select the least loaded host from the available healthy replicas.
+   * Behavior depends on the latencyBasedRoutingEnabled config flag:
+   *
+   * When enabled, uses average response latency as the criterion with a 1.5x spectrum
+   * threshold among the healthy host list. Ref {@link #AVG_LATENCY_SPECTRUM_FOR_HOST_SELECTION}.
+   * Randomly selects within the latency spectrum to avoid skewed traffic.
+   *
+   * When disabled (default), uses pending request count to decide on the host selection
+   * among the healthy host list.
+   *
+   * Note: Healthy host list is based on pending queue count: Ref {@link VeniceHostHealth#isHostHealthy}
+   */
+  private <H> H selectLeastLoadedHost(List<H> hosts, VenicePath path, RouteHttpStats routeHttpStats)
+      throws RouterException {
+    if (latencyBasedRoutingEnabled) {
+      return selectLeastLoadedHostByLatency(hosts, path, routeHttpStats);
+    } else {
+      return selectLeastLoadedHostByPendingCount(hosts, path);
+    }
+  }
+
+  /**
+   * Select host with the lowest pending request count:
+   * This has concerns where a host with low pending count but high latency could be
+   * selected when the QPS is low, leading to suboptimal routing decisions.
+   */
+  private <H> H selectLeastLoadedHostByPendingCount(List<H> hosts, VenicePath path) throws RouterException {
     H minHost = null;
+    long smallestPendingCount = Long.MAX_VALUE;
+
     for (H h: hosts) {
       Instance node = (Instance) h;
-      if (!path.canRequestStorageNode(node.getNodeId()))
+      String nodeId = node.getNodeId();
+
+      if (!path.canRequestStorageNode(nodeId)) {
         continue;
-      long pendingRequestCount = routeHttpRequestStats.getPendingRequestCount(node.getNodeId());
-      if (pendingRequestCount < minCount) {
-        minCount = pendingRequestCount;
+      }
+
+      long currentPendingCount = routeHttpRequestStats.getPendingRequestCount(nodeId);
+      if (currentPendingCount < smallestPendingCount) {
+        smallestPendingCount = currentPendingCount;
         minHost = h;
       }
     }
+
     if (minHost == null) {
-      if (path.isRetryRequest()) {
-        throw RouterExceptionAndTrackingUtils.newRouterExceptionAndTracking(
-            path.getStoreName(),
-            path.getRequestType(),
-            SERVICE_UNAVAILABLE,
-            "Retry request aborted because of slow route for request path: " + path.getResourceName(),
-            RouterExceptionAndTrackingUtils.FailureType.SMART_RETRY_ABORTED_BY_SLOW_ROUTE);
-      } else {
-        throw RouterExceptionAndTrackingUtils.newRouterExceptionAndTracking(
-            path.getStoreName(),
-            path.getRequestType(),
-            SERVICE_UNAVAILABLE,
-            "Could not find ready-to-serve replica for request path: " + path.getResourceName());
-      }
+      throwNoHostAvailableException(path);
```

Comment:
We no longer want to capture the signal whether the route is triggered by an original request vs retry anymore? That sounds useful to me.